pages/en/lb4/Decorator.md

```diff
@@ -0,0 +1,125 @@
+
+- @operation
+
+  Expose a Controller method as a REST API operation. Usage refer to [code](https://github.com/strongloop/loopback-next/blob/debe221ee7eb07b6d6962416b79592c8f13d1e91/packages/core/src/router/metadata.ts#L230-L234)
+
+  *Janny: Need example in doc?*
+
+- @param
+
+  Describe an input parameter of a Controller method. An example can be found in [Writing Controller methods](controller.htm#Writing-Controller-methods), more usage refer to [tsDoc](https://github.com/strongloop/loopback-next/blob/debe221ee7eb07b6d6962416b79592c8f13d1e91/packages/core/src/router/metadata.ts#L252-L272)
+
+  *Janny: Change link to tsDoc page when it's done*
+  
+## Dependency Injection
+
+@inject is a decorator to annotate method arguments for automatic injection by LoopBack IoC container. It can only be applied on a **method parameter** or a **property**. Basiclly it defines a reflector metadata for the target class.
+
+A quick example can be found in [test case](https://github.com/strongloop/loopback-next/blob/9769e85306a8a866d71f133b1d0c013dde0d43c2/packages/context/test/unit/inject.test.ts#L21-L28)
+
+It's been documented as a LoopBack core concept in [Dependency Injection](Dependency-Injection.htm)
+
+*Janny: we miss a list of all metadata keys we can inject*
+
+## Authentication Decorators
+
+- @authenticate
+
+Mark a controller method as requiring authenticated user, it takes in a strategy name. Example refer to [test case](https://github.com/strongloop/loopback-next/blob/9769e85306a8a866d71f133b1d0c013dde0d43c2/packages/authentication/test/acceptance/basic-auth.ts#L104)
+
+## Repository Decorators
+
+### Model decorators
+
+- @model
+
+Define a model in a repository, see [define-models](http://loopback.io/doc/en/lb4/Repositories.html#define-models)
+
+- @property
+
+Define a property in a model, see [define-models](http://loopback.io/doc/en/lb4/Repositories.html#define-models)
+
+### Relation decorators
+
+*Since relation is still in design phase, we only have unit test but no integration example. Refer to [test case](https://github.com/strongloop/loopback-next/blob/9769e85306a8a866d71f133b1d0c013dde0d43c2/packages/repository/src/decorators/relation.ts) for usage*
+
+- @relation
+
+Register a general relation
+
+- @belongsTo @hasOne @hasMany @embedsOne @embedsMany @referencesOne @referencesMany
+
+Register a specific relation
+
+### Repository decorators
+
+- @repository
+
+A Repository is a type of _Service_ that represents a collection of data within a DataSource, the decorator takes in a model class(or its name) and a datasource instance(or its name). For usage, refer to [test case](https://github.com/strongloop/loopback-next/blob/9769e85306a8a866d71f133b1d0c013dde0d43c2/packages/repository/test/unit/decorator/repository.ts#L90)
+
+## Compose and customize decorators
+
+TBD 
+
+- Use chitchat as an example of compose decorators
+- for customize decorators, I expect to create some validation or hook functions as example
```

Comment:
We don't need to show users how to make their own decorators, that's the TypeScript documentation's focus.

Reply:
@kjdelisle it is a requirement I copied from original wiki page, not sure who put it there but seems someone wants it: https://github.com/strongloop/loopback-next/wiki/Using-decorators